recipe/meta.yaml

```diff
@@ -14,9 +18,12 @@ source:
 
 build:
   skip: true  # [win]
-  number: 0
+  number: {{ build }}
+  string: {{ scalar }}_h{{ PKG_HASH }}_{{ build }}
   run_exports:
     - {{ pin_subpackage('slepc', max_pin='x.x') }}
+  track_features:
+    - slepc_complex  # [scalar == "complex"]
```

Comment:
This does not look right. I think It should be `petsc_complex`.

Reply:
@isuruf We need your input here. Do we need to add `track_features` in here? Once PETSc is configured to use let say `scalar=complex`, then all dependent packages are supposed to use complex numbers. 

Should we use `track_features` or rather let the resolver pick the right petsc package via the package  string?

Reply:
Since he seems to busy, let me try to reply:

I think having the `track_features` here is likely not a problem, since it is simply weighing down the `complex` option for slepc. It is possible that this is not needed here, i.e. that if someone does `conda install -c conda-forge slepc` the real variant will be preferred because it is preferred in the `petsc` dependence (this I don't know).

Note that a matching `petsc` variant will be picked in any case, since it is requested explicitly in the requirements.

P.S. As far as I understand, following @isuruf's pushes to the petsc PR we're anyhow providing another "safety net" through giving the `real` variants a higher build number. I.e. even if we were to remove the `track_features` in both packages, the real variants would still be picked over the complex variants by default.

Reply:
Sorry, I was away. Yes, it's good to have `track_features`.